streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -228,9 +228,16 @@ public void setUp() {
     @AfterEach
     public void tearDown() {
         if (thread != null) {
+            // KAFKA-17122 manually stop taskManager if `thread` is not in CREATED state.
+            if (thread.state() != State.CREATED) {
+                thread.taskManager().shutdown(false);
+            }
```

Comment:
Should there be an `else` ?

Reply:
No, for now. We still want to run `thread.shutdown();` event the thread is not in `CREATED` state since it updates the state of the thread as well.